consensus/src/epoch_manager.rs

```diff
@@ -426,6 +429,8 @@ impl EpochManager {
     }
 
     async fn initiate_new_epoch(&mut self, proof: EpochChangeProof) -> anyhow::Result<()> {
+        // The recovery manager exits before starting a new epoch. So, we unset the recovery_mode flag here.
+        self.recovery_mode = false;
```

Comment:
it may be better to make a symmetric assignment in `start_new_epoch`'s two branch

Reply:
Done